### PR TITLE
check codespell & fixing some...

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixing typos and preventing adding new...
Closes #2102 because of https://github.com/Lightning-AI/torchmetrics/pull/2102#issuecomment-1731578783

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2103.org.readthedocs.build/en/2103/

### CODE REVIEW

**Comment by Borda** [comment]
seems ignore does not work properly - https://github.com/codespell-project/codespell/issues/2839#issuecomment-1731987847

File: docs/source/pages/lightning.rst [hunk 1]
```diff
@@ -194,7 +194,7 @@ The following contains a list of pitfalls to be aware of:
 
 * Calling ``self.log("val", self.metric(preds, target))`` with the intention of logging the metric object. Because
   ``self.metric(preds, target)`` corresponds to calling the forward method, this will return a tensor and not the
-  metric object. Such logging will be wrong in this case. Instead it is important to seperate into seperate lines:
+  metric object. Such logging will be wrong in this case. Instead it is important to separate into separate lines:
```
Comment (lucadiliello) [inline docs/source/pages/lightning.rst:197]: maybe `separate into several lines`?

File: src/torchmetrics/aggregation.py [hunk 2]
```diff
@@ -327,18 +327,18 @@ class SumMetric(BaseAggregator):
...
     As output of `forward` and `compute` the metric returns the following output
 
     - ``agg`` (:class:`~torch.Tensor`): scalar float tensor with aggregated sum over all inputs received
 
     Args:
         nan_strategy: options:
-            - ``'error'``: if any `nan` values are encounted will give a RuntimeError
-            - ``'warn'``: if any `nan` values are encounted will give a warning and continue
+            - ``'error'``: if any `nan` values are encountered will give a RuntimeError
+            - ``'warn'``: if any `nan` values are encountered will give a warning and continue
```
Comment (lucadiliello) [inline src/torchmetrics/aggregation.py:339]: give a warning and continue by ignoring or by replacing them with some float?

File: src/torchmetrics/aggregation.py [hunk 3]
```diff
@@ -432,18 +432,18 @@ class CatMetric(BaseAggregator):
...
     As output of `forward` and `compute` the metric returns the following output
 
     - ``agg`` (:class:`~torch.Tensor`): scalar float tensor with concatenated values over all input received
 
     Args:
         nan_strategy: options:
-            - ``'error'``: if any `nan` values are encounted will give a RuntimeError
-            - ``'warn'``: if any `nan` values are encounted will give a warning and continue
+            - ``'error'``: if any `nan` values are encountered will give a RuntimeError
+            - ``'warn'``: if any `nan` values are encountered will give a warning and continue
```
Comment (lucadiliello) [inline src/torchmetrics/aggregation.py:444]: give a warning and continue by ignoring or by replacing them with some float?

File: src/torchmetrics/aggregation.py [hunk 4]
```diff
@@ -496,20 +496,20 @@ class MeanMetric(BaseAggregator):
...
     As output of `forward` and `compute` the metric returns the following output
 
     - ``agg`` (:class:`~torch.Tensor`): scalar float tensor with aggregated (weighted) mean over all inputs received
 
     Args:
        nan_strategy: options:
-            - ``'error'``: if any `nan` values are encounted will give a RuntimeError
-            - ``'warn'``: if any `nan` values are encounted will give a warning and continue
+            - ``'error'``: if any `nan` values are encountered will give a RuntimeError
+            - ``'warn'``: if any `nan` values are encountered will give a warning and continue
```
Comment (lucadiliello) [inline src/torchmetrics/aggregation.py:510]: give a warning and continue by ignoring or by replacing them with some float?

File: src/torchmetrics/aggregation.py [hunk 5]
```diff
@@ -632,10 +632,10 @@ class RunningMean(Running):
     Args:
         window: The size of the running window.
         nan_strategy: options:
-            - ``'error'``: if any `nan` values are encounted will give a RuntimeError
-            - ``'warn'``: if any `nan` values are encounted will give a warning and continue
+            - ``'error'``: if any `nan` values are encountered will give a RuntimeError
+            - ``'warn'``: if any `nan` values are encountered will give a warning and continue
```
Comment (lucadiliello) [inline src/torchmetrics/aggregation.py:636]: give a warning and continue by ignoring or by replacing them with some float?

File: src/torchmetrics/aggregation.py [hunk 6]
```diff
@@ -689,10 +689,10 @@ class RunningSum(Running):
...
```
Comment (lucadiliello) [inline src/torchmetrics/aggregation.py:693]: give a warning and continue by ignoring or by replacing them with some float?

**Comment by Borda** [comment]
@lucadiliello, all good points; mind adding them as suggestions so they are easier to accept?

File: docs/paper_JOSS/paper.bib [hunk 7]
```diff
@@ -66,7 +66,7 @@ @article{scikit_learn
 
 @misc{keras,
     title={Keras},
-    author={Chollet, Fran\c{c}ois and others},
+    author={Chollet, Francois and others},
```
Comment (SkafteNicki) [inline docs/paper_JOSS/paper.bib:69]: this should not be corrected

Comment (Borda) [reply docs/paper_JOSS/paper.bib:69]: unfortunately the `\c{c}` seems to be splitting the word into two and ois is a problem...

File: src/torchmetrics/functional/image/lpips.py [hunk 8]
```diff
@@ -181,13 +181,13 @@ def forward(self, x: Tensor) -> NamedTuple:
...
-    return in_tens.mean([2, 3], keepdim=keepdim)
+def spatial_average(in_tens: Tensor, keep_dim: bool = True) -> Tensor:
+    """Spatial averaging over height and width of images."""
+    return in_tens.mean([2, 3], keepdim=keep_dim)
 
 
-def upsam(in_tens: Tensor, out_hw: Tuple[int, ...] = (64, 64)) -> Tensor:
-    """Upsample input with bilinear interpolation."""
+def upsample(in_tens: Tensor, out_hw: Tuple[int, ...] = (64, 64)) -> Tensor:
+    """Up-sample input with bi-linear interpolation."""
```
[the remark this answers is not in the document]
Comment (Borda) [reply src/torchmetrics/functional/image/lpips.py:190]: pycharm was complaining :)

File: docs/source/pages/lightning.rst (hunk 1 above)
Comment (Borda) [reply docs/source/pages/lightning.rst:197]:
```suggestion
  metric object. Such logging will be wrong in this case. Instead it is important to separate into separate into several lines:
```

File: src/torchmetrics/functional/image/lpips.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (SkafteNicki) [reply src/torchmetrics/functional/image/lpips.py:190]: then stop using pycharm 😃

[the remark this answers is not in the document]
Comment (Borda) [reply src/torchmetrics/functional/image/lpips.py:190]: anyway `upsam` is strange meaningless name :P

[the remark this answers is not in the document]
Comment (SkafteNicki) [reply src/torchmetrics/functional/image/lpips.py:190]: yes, my comment was on the insertion of hyphens in upsample and bilinear, but it does not really matter to me

[the remark this answers is not in the document]
Comment (Borda) [reply src/torchmetrics/functional/image/lpips.py:190]:
```suggestion
    """Upsample input with bilinear interpolation."""
```

File: docs/paper_JOSS/paper.bib (hunk 7 above)
Comment (awaelchli) [reply docs/paper_JOSS/paper.bib:69]: The other option is to get a french keyboard and type the letter with the key that has this character. Or copy it from here: ç

File: docs/source/pages/lightning.rst [hunk 9]
```diff
@@ -194,7 +194,7 @@ The following contains a list of pitfalls to be aware of:
 
 * Calling ``self.log("val", self.metric(preds, target))`` with the intention of logging the metric object. Because
   ``self.metric(preds, target)`` corresponds to calling the forward method, this will return a tensor and not the
-  metric object. Such logging will be wrong in this case. Instead it is important to seperate into seperate lines:
+  metric object. Such logging will be wrong in this case. Instead it is important to separate into separate into several lines:
```
Comment (awaelchli) [inline docs/source/pages/lightning.rst:197]: this repetation looks odd.

File: src/torchmetrics/functional/image/lpips.py [hunk 10]
```diff
@@ -181,12 +181,12 @@ def forward(self, x: Tensor) -> NamedTuple:
         return vgg_outputs(h_relu1_2, h_relu2_2, h_relu3_3, h_relu4_3, h_relu5_3)
 
 
-def spatial_average(in_tens: Tensor, keepdim: bool = True) -> Tensor:
-    """Spatial averaging over heigh and width of images."""
-    return in_tens.mean([2, 3], keepdim=keepdim)
+def spatial_average(in_tens: Tensor, keep_dim: bool = True) -> Tensor:
```
Comment (awaelchli) [inline src/torchmetrics/functional/image/lpips.py:184]: this would be a bc. are you sure?

**Review by awaelchli — APPROVED** [review]

File: docs/paper_JOSS/paper.bib (hunk 7 above)
Comment (awaelchli) [reply docs/paper_JOSS/paper.bib:69]: I think the tool should exclude this because the latex notation is correct here.

Comment (Borda) [reply docs/paper_JOSS/paper.bib:69]: Yes, I am in touch with issues with resolving a few other cases... but still better than nothing :)

Comment (Borda) [reply docs/paper_JOSS/paper.bib:69]:
```suggestion
    author={Chollet, François and others},
```

File: src/torchmetrics/functional/image/lpips.py (hunk 10 above)
Comment (Borda) [reply src/torchmetrics/functional/image/lpips.py:184]: @SkafteNicki can we make this private? :thinking: 
```suggestion
def _spatial_average(in_tens: Tensor, keep_dim: bool = True) -> Tensor:
```

File: docs/source/pages/lightning.rst (hunk 9 above)
Comment (Borda) [reply docs/source/pages/lightning.rst:197]:
```suggestion
  metric object. Such logging will be wrong in this case. Instead it is important to separate into several lines:
```

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**